### PR TITLE
add api endpoint for adding comments

### PR DESCRIPTION
**Pull request by kzzztkt**
ticket #7 
Endpoint for adding comments

### CODE REVIEW

File: __tests__/endpoints.test.js [hunk 1]
```diff
@@ -145,4 +145,24 @@ describe('GET /api/reviews/:review_id/comments', () => {
...
+        return request(app).post('/api/reviews/3/comments')
+        .send(body)
+        .expect(201)
+        .then(({body}) => {
+                expect('author' in body.comment).toBe(true);
+                expect('body' in body.comment).toBe(true);
+                expect('comment_id' in body.comment).toBe(true);
+                expect('created_at' in body.comment).toBe(true);
+                expect('review_id' in body.comment).toBe(true);
+                expect('votes' in body.comment).toBe(true);
```
Comment (yalicee) [inline __tests__/endpoints.test.js]: you can test this with actual data since you know most of it

**Review by yalicee — COMMENTED** [review]
You are missing the following tests:

Status 201, ignores unnecessary properties
Status 400, invalid ID, e.g. string of "not-an-id"
Status 404, non existent ID, e.g. 0 or 9999
Status 400, missing required field(s), e.g. no username or body properties
Status 404, username does not exist

File: __tests__/endpoints.test.js [hunk 2]
```diff
@@ -145,4 +145,76 @@ describe('GET /api/reviews/:review_id/comments', () => {
...
+            username: 'mallionaire',
+            body: 'This is my new comment!'
+          }
+        return request(app).post('/api/reviews/3/comments')
+        .send(body)
+        .expect(201)
+        .then(({body}) => {
+                expect(typeof body.comment.author).toBe('string');
+                expect(typeof body.comment.body).toBe('string');
+                expect(typeof body.comment.comment_id).toBe('number');
```
Comment (B7Bprog) [inline __tests__/endpoints.test.js:162]: You could check the actual comment ID here.

File: __tests__/endpoints.test.js [hunk 3]
```diff
@@ -145,4 +145,76 @@ describe('GET /api/reviews/:review_id/comments', () => {
...
+                expect(typeof body.comment.created_at).toBe('string');
+                expect(typeof body.comment.review_id).toBe('number');
+                expect(typeof body.comment.votes).toBe('number');
+        })
+    });
+    test('should respond: 201 with new comment ', () => {
```
Comment (B7Bprog) [inline __tests__/endpoints.test.js:168]: You said: 'should respond: 201 with new comment', but you're only checking the status code.

File: __tests__/endpoints.test.js [hunk 4]
```diff
@@ -145,4 +145,76 @@ describe('GET /api/reviews/:review_id/comments', () => {
...
+    });
+    test('should respond: 404 from none existant id requests', () => {
+        //Arrange
+        const body = {
+            username: 'mallionaire',
+            body: 'This is my new comment!'
+          }
+        return request(app).post('/api/reviews/99999/comments')
+        .send(body)
+        .expect(404)
```
Comment (B7Bprog) [inline __tests__/endpoints.test.js:198]: Please also expect an error message here, not just a code.

File: __tests__/endpoints.test.js [hunk 5]
```diff
@@ -145,4 +145,76 @@ describe('GET /api/reviews/:review_id/comments', () => {
...
+    });
+    test('should respond: 400 from lack of data', () => {
+        //Arrange
+        const body = {
+            username: 'mallionaire',
+          }
+        return request(app).post('/api/reviews/3/comments')
+        .send(body)
+        .expect(400)
```
Comment (B7Bprog) [inline __tests__/endpoints.test.js:207]: Missing error message.

File: __tests__/endpoints.test.js [hunk 6]
```diff
@@ -145,4 +145,76 @@ describe('GET /api/reviews/:review_id/comments', () => {
...
+    });
+    test('should respond: 404 if username does not exist', () => {
+        //Arrange
+        const body = {
+            username: 'Ben',
+            body: 'This is my new comment!'
+          }
+        return request(app).post('/api/reviews/3/comments')
+        .send(body)
+        .expect(404)
```
Comment (B7Bprog) [inline __tests__/endpoints.test.js:217]: Missing error message.

File: controllers/postComments.controller.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+const { postCommentModel } = require("../models/postComments.model")
+
+exports.postCommentController = (req, res, next) => {
+    const { review_id } = req.params;
+    const comment = req.body;
+    // console.log(review_id);
```
Comment (B7Bprog) [inline controllers/postComments.controller.js:6]: Please remove all commented out code.

File: testDb.txt [hunk 8]
```diff
@@ -0,0 +1,12 @@
+You are now connected to database "nc_games_test" as user "benjohnson".
```
Comment (B7Bprog) [inline testDb.txt:1]: This file shouldn't be part of your PR.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Benjamin, Please see my comments, refactor your code, then run the tests. Once everything passes, feel free to merge it.

**Pull request merged**